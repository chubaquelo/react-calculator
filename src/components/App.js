import React from 'react';
import operate from '../logic/operate';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
// import calculate from '../logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      next: '',
      operation: '',
    };
    this.updateCalcData = this.updateCalcData.bind(this);
    this.handleEqual = this.handleEqual.bind(this);
  }

  updateCalcData = input => {
    const { total, operation, next } = this.state;

    switch (input) {
      case '+':
      case '-':
      case '/':
      case 'x':
      case '%':
        if (parseFloat(total, 10) !== 0) {
          this.setState({ operation: input });
        }
        break;
      case '+/-':
        if (total.includes('-')) {
          this.setState({ total: 'sin -' });
        } else {
          this.setState(prevState => ({ total: '-'.concat(prevState.total) }));
        }
        break;
      case 'AC':
        this.setState({
          total: '0',
          next: '',
          operation: '',
        });
        window.console.log('Set to 0 with AC.');
        break;
      case '1':
      case '2':
      case '3':
      case '4':
      case '5':
      case '6':
      case '7':
      case '8':
      case '9':
      case '0':
        if (operation !== '') {
          this.setState(prevState => ({ next: prevState.next + input }));
        }
        if (total === '0' && operation === '') {
          this.setState({ total: input });
        } else if (next === '' && operation === '') {
          this.setState(prevState => ({ total: prevState.total + input }));
        }
        break;
      case '.':
        if (operation !== '' && !next.includes('.')) {
          this.setState(prevState => ({ next: prevState.next + input }));
        }
        if (total === '0' && operation === '') {
          this.setState({ total: input });
        } else if (
          next === ''
          && operation === ''
          && !total.includes('.')
        ) {
          this.setState(prevState => ({ total: prevState.total + input }));
        }
        break;
      default:
        break;
    }
  };

  handleEqual = () => {
    const { total, operation, next } = this.state;
    let result = '';
    if (
      total !== 0
      && operation !== ''
      && next > 0
    ) {
      result = operate(parseFloat(total, 10), parseFloat(next, 10), operation).toString();
    }
    this.setState({
      total: result,
      operation: '',
      next: '',
    });
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="w-4/5 sm:w-1/2 md:w-1/4 lg:w-1/4 xl:w-1/4">
        <Display total={total} next={next} operation={operation} />
        <ButtonPanel
          updateCalc={this.updateCalcData}
          handleEqual={this.handleEqual}
        />
      </div>
    );
  }
}

export default App;
