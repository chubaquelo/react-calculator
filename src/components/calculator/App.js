import React from 'react';
import operate from './logic/operate';
import ButtonPanel from './ButtonPanel';
import Display from './Display';
import calculate from './logic/calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
      next: '',
      operation: '',
    };
    this.updateCalcData = this.updateCalcData.bind(this);
    this.handleEqual = this.handleEqual.bind(this);
  }

  updateCalcData = btnName => {
    const input = this.state;
    this.setState(calculate(input, btnName));
  }

  handleEqual = () => {
    const { total, operation, next } = this.state;
    let result = '';
    if (
      total !== 0
      && operation !== ''
      && next > 0
    ) {
      result = operate(parseFloat(total, 10), parseFloat(next, 10), operation).toString();
    }
    this.setState({
      total: result,
      operation: '',
      next: '',
    });
  };

  render() {
    const { total, next, operation } = this.state;
    return (
      <div className="w-screen text-center flex justify-center">
        <div className="w-4/5 sm:w-1/2 md:w-1/4 lg:w-1/4 xl:w-1/4">
          <Display total={total} next={next} operation={operation} />
          <ButtonPanel
            updateCalc={this.updateCalcData}
            handleEqual={this.handleEqual}
          />
        </div>
      </div>
    );
  }
}

export default App;
