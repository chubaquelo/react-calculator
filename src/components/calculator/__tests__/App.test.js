import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('Test App Snapshot', () => {
  it('matches App snapshot', () => {
    const forTest = renderer.create(<App />).toJSON();
    expect(forTest).toMatchSnapshot();
  });
});
