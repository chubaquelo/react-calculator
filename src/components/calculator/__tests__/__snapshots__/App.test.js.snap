// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App Snapshot matches App snapshot 1`] = `
<div
  className="w-screen text-center flex justify-center"
>
  <div
    className="w-4/5 sm:w-1/2 md:w-1/4 lg:w-1/4 xl:w-1/4"
  >
    <div
      className="break-all text-right bg-black text-white text-5xl w-full h-18 p-2 flex justify-end items-end rounded-t-lg border-white border-4 border-b-0 select-all"
    >
      0
    </div>
    <div
      className="w-full border-white border-4 border-b-0 flex flex-wrap"
    >
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="w-full border-white border-4 border-b-0 flex flex-wrap"
    >
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="w-full border-white border-4 border-b-0 flex flex-wrap"
    >
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="w-full border-white border-4 border-b-0 flex flex-wrap"
    >
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="w-full border-white border-4 flex flex-wrap"
    >
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl w-2/4 text-3xl"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl bg-blue-400 hover:bg-blue-900"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
